metaflow/includefile.py

```diff
@@ -17,11 +20,41 @@ def __init__(self, logger, is_text, encoding, path):
         self._is_text = is_text
         self._encoding = encoding
         self._path = path
-        self._size = os.path.getsize(self._path)
+        self._size = 0
 
-    def __call__(self):
+    @classmethod
+    def is_file_handled(cls, path):
+        if path.startswith("s3://"):
+            return True, None
+        try:
+            with open(path, mode='r') as _:
+                pass
+        except OSError:
+            return False, "Could not open file '%s'" % path
+        else:
+            return True, None
+
+    def handle_s3_file(self):
```

Comment:
Can these be private methods? So _handle_s3_file since I don't think you need them accessible outside.